jobbergate-api/jobbergate_api/apps/job_submissions/routers.py

```diff
@@ -313,7 +315,24 @@ async def job_submissions_agent_pending(
     )
     logger.trace(f"query = {render_sql(query)}")
     rows = await database.fetch_all(query)
-    return rows
+
+    try:
+        response = [
+            PendingJobSubmission(
+                **row,
+                job_script_data_as_string=s3man_jobscripts[str(row.get("job_script_id"))],
+            )
+            for row in rows
+        ]
+    except KeyError:
+        message = "JobScript file not found."
+        logger.warning(message)
```

Comment:
This should be an error. Also, include the ids the were not found in the error message.

Reply:
solved in [90e2940](https://github.com/omnivector-solutions/jobbergate/pull/130/commits/90e29406e5fd144a5c5b394c5cf113140e03d182)